### ISSUE TITLE
[issue]: Windows 7 - not booting usb3.x drivers...

### ISSUE THREAD

**Issue by PaulNevada** [ventoy/Ventoy#1327]
### Official FAQ

- [X] I have checked the official FAQ.

### Ventoy Version

1.0.44

### What about latest release

No. I didn't try the latest release.

### BIOS Mode

Both

### Partition Style

MBR

### Disk Capacity

256GB

### Disk Manufacturer

Samsung

### Image file checksum (if applicable)

Yes.

### What happened?

Hi!
I am not sure it's a Ventoy's bug really. But, please, explain to me - why does it happen with the 50/50 - sometimes my usb-drive(s - even!) doesn't see the usb-port and sometimes it does? It fails after I proceed to the Windows Edition Selection Menu in its typical installation process...

**Comment by steve6375** [comment]
Do you mean if you legacy boot the same ventoy usb drive on the same target system from the same usb port, from a power switch-on, sometimes it sees the usb drive and on some boots it does not?

**Comment by PaulNevada** [comment]
The setup sees the drive always, but it starts to loose it when I proceed from the Step1, so to speak, and move to the next screens. Then I receive "The driver is missing", but, again in some weird 50/50 cases...

**Comment by steve6375** [comment]
Well windows 7 is old. It does not contain usb drivers for modern systems and usb 3 had not even been invented!
What do you mean by 50/50 cases? What cases do you mean? What is case  A and what is case B?

**Comment by PaulNevada** [comment]
The same environment in everything - as you listed above. I use the Windows 7 iso that does include all the needed usb 3.x drivers (modifies one). I see it as 50/50 in the behavior...

**Comment by steve6375** [comment]
So your answer to my original question is YES?
Cold power on each time, same usb port, same system, same usb drive?

**Comment by PaulNevada** [comment]
"Cold power" needs to be checked. Maybe, it is...

**Comment by steve6375** [comment]
If you are using a usb 3 drive in a usb 3 port then the added win7 usb drivers will be used to access the iso file on the usb 3 drive. Maybe they driver does not detect the drive quick enough? Ventoy injects code which finds the usb drive and then loads the iso as a virtual dvd drive when winpe is loaded. So there could be a timing problem?

**Comment by PaulNevada** [comment]
Wait, I said 50/50 because I'm pretty stuck with that. If I reboot the pc it works (all the?) time. If I cold start - dunno - will check it again. I thought it's kinda randomized here...

**Comment by PaulNevada** [comment]
I am pretty sure - yes - this is a timing thing! Is the a way to extend this parameter for me and without the compiling the code?

**Comment by steve6375** [comment]
I don't think so. Not sure if the code has a retry algorithm or any built in delay. The developer will have to respond.

**Comment by steve6375** [comment]
Maybe using a usb 2 port or usb 2 extension cable will solve the issue?

**Comment by PaulNevada** [comment]
I do want to be fixed. Even by myself Have some _little_ programming lnowledges...

**Comment by PaulNevada** [comment]
The comments form the developer would be very helpful - for sure. Thanks for the help!

**Comment by PaulNevada** [comment]
Tried rufus several times with the current environment - the same strange behavior...

**Comment by steve6375** [comment]
With Rufus, you may be able to remove the USB drive and re-insert it once running Setup. It may then re-detect the USB drive.
With both Ventoy and Rufus you can press SHIFT+F10 once Setup is running to get a command console - then type NOTEPAD and use the File - Open dialogue to see if you can 'see' the USB drive - if not then the USB drive in the particular ISO that you are using is not working properly. This issue is not a problem with Rufus or Ventoy - it is a problem with the added USB drivers inside the ISO.

**Comment by NewbieXvwu** [comment]
You should install a USB 3.0 driver to use a USB disk that uses USB 3.0 on Windows 7.
You can try to install Windows 7 in a PE(Preinstall Envirionment).Most of them are installed USB 3.0 driver.

**Comment by csdvrx** [comment]
There are a few such images around if you are in a hurry, google for Windows7U_x64_SP1_ZEN_v1 as the AMD Zen CPUs have similar issues.

BTW here's a link to a post explaining how to do that the legal way, with the official image and DISM: https://community.hwbot.org/topic/166612-windows-7-x64-sp1-with-integrated-drivers-for-zen/
